internal/app/cli.go

```diff
@@ -218,32 +247,51 @@ func (c *cli) readState(s *state) error {
 	os.RemoveAll(tempFilesDir)
 	_ = os.MkdirAll(tempFilesDir, 0o755)
 
+	if len(c.spec) > 0 {
+
+		sp := new(StateFiles)
+		sp.specFromYAML(c.spec)
+
+		for _, val := range sp.StateFiles {
+			fo := fileOption{}
+			fo.name = val.Path
+			fo.priority = val.Priority
+			checkSpecValid(fo.name)
+			c.files = append(c.files, fo)
+		}
+	}
+
 	// read the TOML/YAML desired state file
 	for _, f := range c.files {
 		var fileState state
 
-		if err := fileState.fromFile(f); err != nil {
+		if err := fileState.fromFile(f.name); err != nil {
 			return err
 		}
-		log.Infof("Parsed [[ %s ]] successfully and found [ %d ] apps", f, len(fileState.Apps))
+
+		if f.priority != 0 {
+			fileState.patchPriority(f.priority)
+		}
```

Comment:
This might have some edge cases and I think it would be better to stable sort the list of files based on their priority before this loop.